src/Sulu/Bundle/WebsiteBundle/Twig/Content/ContentPathTwigExtension.php

```diff
@@ -84,6 +84,16 @@ public function getContentPath($route, $webspaceKey = null, $locale = null, $dom
             $domain,
             $scheme
         );
+        if ($allDomains && !$url) {
```

Comment:
currently the behaviour is that it returns the route itself if the given domain does not matches it. with this parameter true it tries to return any domain if the "prefered" one is not found.